lfsapi/client.go

```diff
@@ -76,6 +77,37 @@ func (c *Client) Do(req *http.Request) (*http.Response, error) {
 	return res, c.handleResponse(res)
 }
 
+func (c *Client) extraHeadersFor(req *http.Request) http.Header {
+	copy := make(http.Header, len(req.Header))
```

Comment:
You could even get the values from `c.extraHeaders()` to get a more accurate initial size :)